kommunicate/src/main/java/io/kommunicate/network/SSLPinningConfig.java

```diff
@@ -0,0 +1,74 @@
+package io.kommunicate.network;
+
+import static com.google.gson.internal.$Gson$Types.arrayOf;
+
+import java.security.KeyManagementException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Objects;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+public class SSLPinningConfig {
+
+    public synchronized static SSLSocketFactory createPinnedSSLSocketFactory() {
+        String expectedPublicKeyHash = "";
+
+        TrustManager trustManager = new X509TrustManager() {
+            @Override
+            public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+                // No client side authentication implemented
+            }
+
+            @Override
+            public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+                if (chain.length == 0) {
+                    throw new CertificateException("Certificate chain is empty");
+                }
+
+                X509Certificate certificate = chain[0];
+                PublicKey publicKey = certificate.getPublicKey();
+                byte[] publicKeyBytes = publicKey.getEncoded();
+
+                MessageDigest md = null;
+                try {
+                    md = MessageDigest.getInstance("SHA-256");
+                } catch (NoSuchAlgorithmException e) {
+                    e.printStackTrace();
+                    throw new RuntimeException(e);
+                }
+                byte[] publicKeyHash = md.digest(publicKeyBytes);
+                String publicKeyHashBase64 = android.util.Base64.encodeToString(publicKeyHash, android.util.Base64.NO_WRAP);
+
+                if (!Objects.equals(publicKeyHashBase64, expectedPublicKeyHash)) {
```

Comment:
Change it to support multiple regions

Reply:
Done,

Reply:
Used the secrets library to store the hases and used it as build config. Because to avoid store the hashes publicly in  the opensource repo.